lms/djangoapps/sneakpeek_deeplink/tests.py

```diff
@@ -143,3 +155,18 @@ def test_sneakpeek_course_enrollment_closed(self):
         req.path = '/courses/closed/course/run/info'
         self.assertIsNone(self.middleware.process_request(req))
         self.assertNoSneakPeek(req, self.closed_course)
+
+    def test_deleted_course_with_preferences(self):
+        # initial sneakpeek for the created course should be successful
+        initial_req = self.make_successful_sneakpeek_login_request()
```

Comment:
Honestly, I'd probably just reuse the variable `request` throughout here.
It makes it easier to step through code while debugging.

```python
request = do_successfull_request()
# ...
request = do_failing_request()
# ...
```